### PR TITLE
Fixes #6831 - expire topbar cache for admin on taxonomy updates

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

**Comment by domcleal** [comment]
There are quite a few test failures @tbrisker.

File: app/models/taxonomy.rb [hunk 1]
```diff
@@ -126,6 +126,7 @@ def dup
 
   def expire_topbar_cache(sweeper)
     users.each { |u| u.expire_topbar_cache(sweeper) }
+    User.current.expire_topbar_cache(sweeper) if User.current.admin
```
Comment (domcleal) [inline app/models/taxonomy.rb]: What about other users who are logged in, this won't clear their caches?

**Comment by ohadlevy** [comment]
looks like random test failures [test]

**Comment by domcleal** [comment]
There were some caused by the change that @tbrisker fixed.  (It'd be great if you could leave a comment to say what's changed, so we know to come back to the PR, thanks.)

File: app/models/taxonomy.rb (hunk 1 above)
Comment (ohadlevy) [inline app/models/taxonomy.rb]: how about something like

``` ruby
(User.admin + users).each {|u| u.expire_topbar_cache(sweeper) }
```

Comment (domcleal) [inline app/models/taxonomy.rb]: We had that in the first commit, I don't understand why it was reverted.

Comment (tbrisker) [inline app/models/taxonomy.rb]: It caused the tests to fail because it expired more user's caches then was expected. Might be a problem with the tests?

Comment (domcleal) [inline app/models/taxonomy.rb]: The expectations probably need updating to account for the number of admin users plus the users in the taxonomy.

**Comment by tbrisker** [comment]
Ok, changed it back to users+Users.only_admin and corrected the tests to count the admins correctly.

**Comment by domcleal** [comment]
Thanks @tbrisker, merged as 8fabe66 for Foreman 1.6.0.

**Pull request closed without merging**